### PR TITLE
forms global css

### PR DESCRIPTION
**Pull request by sandikbarr**
#27 the global form styles

### CODE REVIEW

File: src/demo/app/global-css/button-demo.component.ts [hunk 1]
```diff
@@ -0,0 +1,151 @@
...
+      <li><code>padding: 0</code></li>
+      <li>The button has <code>margin: @margin-base</code> of 10px but the pseudo <code>::before</code> element has no margins</li>
+      <li>The font icon has <code>color: @blue</code> and on <code>:hover color: @blue-emphasis</code></li>
+      <li>When disabled, the button has <code>color: @grey-silver</code></li>
+      <li>TODO: Show all the available icon examples. We do not yet have icon font support here.</li>
+    </ul>
+    <aside class="example">
+      Example:
+      <pre class="code">&lt;button class="btn-icon icon-cancel"&gt;Click Me&lt;/button&gt;</pre>
+      <button class="btn-icon" aria-hidden="true"><!-- TODO: need icon font support --></button>
```
Comment (sandikbarr) [inline src/demo/app/global-css/button-demo.component.ts]: TODO: We don't have icon font support yet

File: src/demo/app/global-css/layout-demo.component.ts [hunk 2]
```diff
@@ -3,21 +3,24 @@ import {Component} from '@angular/core';
...
-      <li>Open Sans, sans-serif font-family on <code>body</code></li>
-      <li>1.5em line-height on <code>html</code></li>
-      <li>1.5rem line-height on <code>body</code></li>
-      <li>White smoke background on <code>main</code></li>
-      <li>Top padding on <code>main</code> to make space for header nav</li>
-      <li>1060px max-width on the <code>.main section</code> with additional horizontal margin and padding</li>
-    </ul>
+    <h1>CSS layout</h1>
+    <p>A style sheet used to create a global layout</p>
+    <section>
```
Comment (sandikbarr) [inline src/demo/app/global-css/layout-demo.component.ts]: mostly just added this `section` wrapper

File: src/demo/app/global-css/reset-demo.component.ts [hunk 3]
```diff
@@ -3,84 +3,87 @@ import {Component} from '@angular/core';
...
-        <code>font-family: monospace, sans-serif</code> for 
-        <code>code</code>, <code>kbd</code>, <code>pre</code>, and <code>samp</code>
-      </li>
-      <li>Provide <code>.ie7</code> <code>-ms-interpolation-mode: bicubic</code> <a href="ms-interpolation-mode: bicubic">fix</a></li>
-      <li>Provide <code>.clearfix</code> for clearing floats</li>
-    </ul>
+    <h1>CSS reset</h1>
+    <p>A style sheet designed to strip initial styles from browsers to help keep designs consistent</p>
+    
+    <section>
```
Comment (sandikbarr) [inline src/demo/app/global-css/reset-demo.component.ts]: mostly just added `section` wrappers here

File: src/lib/styles/forms.less [hunk 4]
```diff
@@ -1,50 +1,136 @@
...
   vertical-align: baseline;
   background: 0 0;
+  outline: none;
+  cursor: pointer;
 }
 
-input[type=radio] {
-  vertical-align: text-bottom;
+label + textarea {
+  vertical-align: top;
```
Comment (sandikbarr) [inline src/lib/styles/forms.less]: This is a new rule, but we aren't using textarea anywhere. It just looks better with the horizontal form layout.

File: src/lib/styles/modal.less [hunk 5]
```diff
@@ -0,0 +1,37 @@
+/* TODO: create separate modal file */
+/* Seems like we maybe could use this modal stuff for projected content in the modal but we're not right now. Should I keep it? */
```
Comment (sandikbarr) [inline src/lib/styles/modal.less]: This isn't documented, and I don't think it's used. Seems like it's a remnant from beta.

File: src/lib/styles/variables.less [hunk 6]
```diff
@@ -3,7 +3,9 @@
 @grey-dark: #343434;
 @grey-dove: #676767;
 @grey: #818181;
+@grey-suva: #939393;
```
Comment (sandikbarr) [inline src/lib/styles/variables.less]: suva is actually a thing http://www.htmlcsscolor.com/hex/939393

File: src/lib/styles/variables.less [hunk 7]
```diff
@@ -3,7 +3,9 @@
...
 @grey-silver: #a3a3a3;
+@grey-northernsky: #bebebe;
```
Comment (sandikbarr) [inline src/lib/styles/variables.less]: This one is made up. It's rarely used, so it's got a reference so obscure you'll wonder why you're using it. BEBEBE is how Nick Drake tunes his guitar.

File: src/lib/styles/forms.less [hunk 8]
```diff
@@ -1,50 +1,136 @@
...
+  flex: 1 1 215px;
+  flex-basis: 215px;
+  -ms-flex-preferred-size: 215px;
+  -webkit-flex-basis: 215px;
 }
 
-input, select, textarea {
-  font: 99% sans-serif
+::placeholder {
+  color: @grey-northernsky;
```
Comment (sandikbarr) [inline src/lib/styles/forms.less]: new rule for placeholders

File: src/lib/styles/modal.less (hunk 5 above)
Comment (cavis) [inline src/lib/styles/modal.less]: Meaning that our current modals are unstyled?

File: src/lib/styles/variables.less (hunk 7 above)
Comment (cavis) [inline src/lib/styles/variables.less]: Nice.

File: src/lib/styles/modal.less (hunk 5 above)
Comment (sandikbarr) [inline src/lib/styles/modal.less]: the stuff we're using on the Publish modal looks like it is styled with deep selectors

**Pull request merged**